ui/app/src/context/DarkMode.tsx

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { createContext, useContext, useMemo } from 'react';
+import { CssBaseline, ThemeProvider, useMediaQuery } from '@mui/material';
+import { getDarkTheme, getLightTheme } from '../theme';
+import { useLocalStorage } from '../utils/browser-storage';
+
+const DARK_MODE_PREFERENCE_KEY = 'PERSES_ENABLE_DARK_MODE';
+
+interface DarkModeContext {
+  isDarkModeEnabled: boolean;
+  setDarkMode: (pref: boolean) => Promise<void>;
+}
+
+export const DarkModeContext = createContext<DarkModeContext>({
+  isDarkModeEnabled: false,
+  setDarkMode: () => {
+    throw new Error('Error: Dark mode not implemented');
+  },
+});
```

Comment:
Minor suggestion: I hate that React makes you define a "default" context implementation like this where, for example, the `setDarkMode` doesn't really do anything. In my opinion, for contexts like this where a default doesn't really work, we should be throwing if the Provider is missing not giving them a default value.

What I've been doing instead is defining the content as `| undefined` and then exporting a hook for using it that does the check for `undefined` and throws a helpful error message. Would be cool to follow that pattern here too. For example:

https://github.com/perses/perses/blob/1f420c168d062bd9bffdae84da15ee1145247811/ui/plugin-system/src/runtime/time-range.ts#L21-L32